daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -333,6 +333,32 @@ private[lf] object Speedy {
       ptx.contractState.locallyCreated.contains(contractId)
     }
 
+    private[speedy] def needPackage(
+        packageId: PackageId,
+        ref: language.Reference,
+        k: () => Control[Question.Update],
```

Comment:
I may be missing something because I am not very familiar with this code yet, but I think that the return type of `k` can be polymorphic:

```
private[speedy] def needPackage[A](
        packageId: PackageId,
        ref: language.Reference,
        k: () => Control[A],
): Control[Question.Update]
```

It won't change anything for this PR but maybe it will come in handy some other time?

Reply:
It cannot be polymorphic, as  `Question.Update.NeedPackage` is of type `Question.Update`

Reply:
I didn't realize that the type of `setControl` was constrained by the context, sorry for the noise.

Reply:
please don't be sorry.